pysat/_meta.py

```diff
@@ -764,163 +534,387 @@ def match_name(func, name, names):
                 if new_key in self.keys_nD():
                     meta_row.at['children'] = self.ho_data[new_key].copy()
                 else:
-
-                    meta_row.at['children'] = None  # empty_meta
+                    meta_row.at['children'] = None  # Return empty meta instance
 
                 return meta_row
             else:
                 raise KeyError('Key not found in MetaData')
         else:
-            raise NotImplementedError("No way to handle MetaData key {}".format(
-                key.__repr__()))
+            raise NotImplementedError("".join(["No way to handle MetaData key ",
+                                               "{}; ".format(key.__repr__()),
+                                               "expected tuple, list, or str"]))
 
-    def _label_setter(self, new_label, current_label, attr_label,
-                      default=np.nan, use_names_default=False):
-        """Generalized setter of default meta attributes
+    # QUESTION: DOES THIS NEED TO CHANGE???
+    def __contains__(self, other):
+        """case insensitive check for variable name
```

Comment:
@rstoneback I am not sure what this routine is supposed to do.  Is it behaving correctly?

Reply:
> Ok @rstoneback, thanks to your insight I was able to make the suggested change at (what I think) is the appropriate level in the classes. I also went through and fixed other stray bugs. Now, all unit tests are passing but 11 in `test_meta`, which are related to a single problem. When you update the labels (e.g., meta.labels.units = 'Units'), the change isn't propagated through to the pandas Series. I'm not sure how this was accomplished before.
> 


Check out the old meta method self._label_setter. I pointed to the general area in the code highlight a few comments down.

Reply:
> @rstoneback I am not sure what this routine is supposed to do. Is it behaving correctly?

The method lets you know if the string provided in other is a variable in the meta object. The string comparison is case insensitive. Other than the docstring looks ok to me.

```
In [8]: In [2]: test = pysat.Instrument('pysat', 'testing')                                                                                                                                                                 
   ...:  
   ...: In [3]: test.load(2009, 1)                                                                                                                                                                                          
   ...:                                                                                                                                                                                                                     

In [9]: 'uts' in test.meta                                                                                                                                                                                                  
contains!
Out[9]: True

In [10]: 'UTS' in test.meta                                                                                                                                                                                                 
contains!
Out[10]: True

In [11]: test.variables                                                                                                                                                                                                     
Out[11]: 
Index(['uts', 'mlt', 'slt', 'longitude', 'latitude', 'altitude', 'orbit_num',
       'dummy1', 'dummy2', 'dummy3', 'dummy4', 'string_dummy', 'unicode_dummy',
       'int8_dummy', 'int16_dummy', 'int32_dummy', 'int64_dummy'],
      dtype='object')
```